### ISSUE TITLE
Train on images, and use on videos

### ISSUE THREAD

**Issue by ben-jy** [facebookresearch/sam3#480]
Hi !
First of all, thank you for the amazing work :)
I have a question regarding the architecture of SAM3 and its training. 
When I look at the diagram you provided, I have the intuition that the new Detector module is responsible of an independent task, which is detecting masks from (text) prompts and image embedding. If it is really the case, would it possible, and make sense, to fine-tune SAM3 on images only, to update weights of the text encoder and detector only, and use/plug these updated modules for video inference, along with the tracker and memory bank ? Or should the finetuning of tracker and memory bank also be considered ?
In my use case, we do not have labeled videos yet, only images, so it would make sense to train SAM3 on images only, and be able to still infer efficiently on videos.
I hope my question is clear enough, and I thank you in advance for your answer.

**Comment by wangxinxin0422** [comment]
have run into the same issue. Would this be a valid solution?

**Comment by heyoeyo** [comment]
> have run into the same issue. Would this be a valid solution?

Yes it should be fine as long as you don't update the weights of the [ViT](https://github.com/facebookresearch/sam3/blob/86ed77094094e5cabb16b0414ec60c5ba9ce0a0f/sam3/model/vitdet.py#L618) (e.g. vision backbone) or the [sam2_convs](https://github.com/facebookresearch/sam3/blob/86ed77094094e5cabb16b0414ec60c5ba9ce0a0f/sam3/model/necks.py#L121-L125) part of the ViT neck. Though training this way won't have a _direct_ affect on the video tracking mask quality/outputs, only the detector results.
